cmd/pebble/cmd_run.go

```diff
@@ -36,34 +37,92 @@ var longRunHelp = `
 The run command starts pebble and runs the configured environment.
 `
 
+var symbolicShutdownSignals = map[string]os.Signal{
+	"TERM": syscall.SIGTERM,
+	"INT":  syscall.SIGINT,
+	"QUIT": syscall.SIGQUIT,
+	"USR1": syscall.SIGUSR1,
+	"USR2": syscall.SIGUSR2,
+}
+
+// defaultShutdownSignals is the list of signals listened to by default for
+// initiating shut down procedures.
+var defaultShutdownSignals = []os.Signal{
+	syscall.SIGTERM,
+	syscall.SIGINT,
+	syscall.SIGQUIT,
+}
+
 type cmdRun struct {
 	clientMixin
 
-	CreateDirs bool   `long:"create-dirs"`
-	Hold       bool   `long:"hold"`
-	HTTP       string `long:"http"`
-	Verbose    bool   `short:"v" long:"verbose"`
+	CreateDirs      bool   `long:"create-dirs"`
+	Hold            bool   `long:"hold"`
+	HTTP            string `long:"http"`
+	ShutdownSignals string `long:"shutdown-signals"`
+	Verbose         bool   `short:"v" long:"verbose"`
 }
 
 func init() {
 	addCommand("run", shortRunHelp, longRunHelp, func() flags.Commander { return &cmdRun{} },
 		map[string]string{
-			"create-dirs": "Create pebble directory on startup if it doesn't exist",
-			"hold":        "Do not start default services automatically",
-			"http":        `Start HTTP API listening on this address (e.g., ":4000")`,
-			"verbose":     "Log all output from services to stdout",
+			"create-dirs":      "Create pebble directory on startup if it doesn't exist",
+			"hold":             "Do not start default services automatically",
+			"http":             `Start HTTP API listening on this address (e.g., ":4000")`,
+			"shutdown-signals": "Symbolic signal(s) pebble listens to for shutting down. Defaults to TERM,INT & QUIT ",
+			"verbose":          "Log all output from services to stdout",
 		}, nil)
 }
 
+func processShutdownSignals(rawUserSignals string) ([]os.Signal, error) {
+	sigs := strings.Split(rawUserSignals, ",")
+	signals := make([]os.Signal, len(sigs))
+	i := 0
+	for _, sig := range sigs {
+		sysSig, exists := symbolicShutdownSignals[sig]
+		if !exists {
+			return signals, fmt.Errorf("unrecognised symbolic signal %q, valid  values: %v",
+				sig, validSymbolicSignals())
+		}
+		signals[i] = sysSig
+	}
+	return signals, nil
+}
+
+func validSymbolicSignals() []string {
+	keys := make([]string, len(symbolicShutdownSignals))
+
+	i := 0
+	for k := range symbolicShutdownSignals {
+		keys[i] = k
+		i++
+	}
+	return keys
+}
+
 func (rcmd *cmdRun) Execute(args []string) error {
 	if len(args) > 0 {
 		return ErrExtraArgs
 	}
 
+	var err error
+	shutdownSignals := defaultShutdownSignals
+	if rcmd.ShutdownSignals != "" {
```

Comment:
Once we're using `default:`, how about we remove this `if` so that empty string is supported and means "no shutdown signals"?

Reply:
Think we have this now.